pkg/ingester/instance.go

```diff
@@ -133,37 +137,59 @@ func (i *instance) consumeChunk(ctx context.Context, ls labels.Labels, chunk *lo
 }
 
 func (i *instance) Push(ctx context.Context, req *logproto.PushRequest) error {
+	record := recordPool.GetRecord()
+	record.UserID = i.instanceID
+	defer recordPool.PutRecord(record)
+
 	i.streamsMtx.Lock()
 	defer i.streamsMtx.Unlock()
 
 	var appendErr error
 	for _, s := range req.Streams {
 
-		stream, err := i.getOrCreateStream(s)
+		stream, err := i.getOrCreateStream(s, false, record)
```

Comment:
why no lock ? I thought many goroutines can push to an instance ?

Reply:
It's locked once for _all_ the streams at line 144 (a few lines above). This is an improvement over locking the streamsMtx for each series.

Reply:
👍